### PR TITLE
Add ChefRuby/RequireNetHttps

### PR DESCRIPTION
**Pull request by tas50**
Catch anywhere we're still doing this.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: config/chefstyle.yml [hunk 1]
```diff
@@ -650,4 +650,9 @@ ChefRuby/GemspecRequireRubygems:
   Enabled: true
   VersionAdded: '1.3.0'
   Include:
-    - '**/*.gemspec'
\ No newline at end of file
+    - '**/*.gemspec'
+
+ChefRuby/RequireNetHttps:
+  Description: net/http is deprecated and just includes net/http and openssl. We should include those directly instead
```
Comment (phiggins) [inline config/chefstyle.yml:656]:
```suggestion
  Description: net/https is deprecated and just includes net/http and openssl. We should include those directly instead
```

?

Comment (phiggins) [reply config/chefstyle.yml:656]: I won't bother commenting on the 5 other places this string appears in this PR

Comment (tas50) [reply config/chefstyle.yml]: fixed

**Comment by phiggins** [comment]
I could see this adding unnecessary requires if someone blindly accepted the auto correction on code like:

```
require "net/http"
require "net/https"
```

That could be way more trouble than its worth to try to guard against though.

**Comment by tas50** [comment]
We have another rubocop to enable in the near future that solves the duplicate issue.

**Pull request merged**